dashboard/config/environments/test.rb

```diff
@@ -100,4 +100,9 @@
   end
 
   config.experiment_cache_time_seconds = 0
+
+  # Prevent merge conflicts on schema.rb by skipping regeneration of schema.rb
+  # on the test machine. this is necessary because as of April 2025 the test DB
+  # schema differs from other environments due to utf8mb3 vs utf8mb4 issues.
+  config.active_record.dump_schema_after_migration = !CDO.chef_managed
```

Comment:
We have a [method](https://github.com/code-dot-org/code-dot-org/blob/847be9fbe3afc1165c9a80e22792d2f0c68b4660/lib/cdo.rb#L296-L301) that is intended to be a more explicit test for whether we’re currently executing on the Test Server ™. 
```suggestion
  config.active_record.dump_schema_after_migration = !CDO.test_system?
```